### PR TITLE
feat: velodyne lidar azimuth correction

### PR DESCRIPTION
**Pull request by ike-kazu**
## PR Type

- Improvement

## Description
I fixed https://github.com/tier4/nebula/pull/112 azimuth correction function pr. It was reverted. 
This pr makes nabula velodyne detecoder lighter.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets

## CI Checks

- **Build and test for PR**: Required to pass before the merge.

### CODE REVIEW

File: nebula_decoders/include/nebula_decoders/nebula_decoders_velodyne/decoders/vlp32_decoder.hpp [hunk 1]
```diff
@@ -52,6 +52,7 @@ class Vlp32Decoder : public VelodyneScanDecoder
   float sin_rot_table_[ROTATION_MAX_UNITS];
   float cos_rot_table_[ROTATION_MAX_UNITS];
   float rotation_radians_[ROTATION_MAX_UNITS];
+  // float vlp_32_laser_azimuth_cache_[16];
```
Comment (mojomex) [inline nebula_decoders/include/nebula_decoders/nebula_decoders_velodyne/decoders/vlp32_decoder.hpp:55]: Please remove this unused line

File: nebula_decoders/src/nebula_decoders_velodyne/decoders/vlp16_decoder.cpp [hunk 2]
```diff
@@ -250,10 +255,8 @@ void Vlp16Decoder::unpack(const velodyne_msgs::msg::VelodynePacket & velodyne_pa
                 const float cos_rot_correction = corrections.cos_rot_correction;
                 const float sin_rot_correction = corrections.sin_rot_correction;
```
Comment (mojomex) [inline nebula_decoders/src/nebula_decoders_velodyne/decoders/vlp16_decoder.cpp:256]: `cos_rot_correction` and `sin_rot_correction` are unused now, please remove them.

**Review by mojomex — CHANGES_REQUESTED** [review]
@ike-kazu thank you for the PR, great work!
There's a few minor things I'd like you to change, mostly removing debug comments.

I still have to test the decoder with a real sensor.

File: nebula_decoders/src/nebula_decoders_velodyne/decoders/vls128_decoder.cpp [hunk 3]
```diff
@@ -252,7 +252,6 @@ void Vls128Decoder::unpack(const velodyne_msgs::msg::VelodynePacket & velodyne_p
           float azimuth_corrected_f = azimuth + (azimuth_diff * vls_128_laser_azimuth_cache_[firing_order]) - corrections.rot_correction * 180.0 / M_PI * 100;
           
           if (azimuth_corrected_f < 0.0){
-            // std::cout << "azimuth_corrected_f: " << azimuth_corrected_f << std::endl;
             azimuth_corrected_f += 36000.0;
           }
           // const int16_t azimuth_corrected = ((int16_t)round(azimuth_corrected_f)) % 36000;
```
Comment (mojomex) [inline nebula_decoders/src/nebula_decoders_velodyne/decoders/vls128_decoder.cpp:257]: This one as well please!

**Review by mojomex — CHANGES_REQUESTED** [review]

File: nebula_decoders/src/nebula_decoders_velodyne/decoders/vls128_decoder.cpp (hunk 3 above)
Comment (ike-kazu) [reply nebula_decoders/src/nebula_decoders_velodyne/decoders/vls128_decoder.cpp]: I forgot this. I fixed it right now. Thank you!

File: nebula_decoders/src/nebula_decoders_velodyne/decoders/vlp16_decoder.cpp [hunk 4]
```diff
@@ -228,10 +228,15 @@ void Vlp16Decoder::unpack(const velodyne_msgs::msg::VelodynePacket & velodyne_pa
...
-              const float azimuth_corrected_f =
+              float azimuth_corrected_f =
                 azimuth +
                 (azimuth_diff * ((dsr * VLP16_DSR_TOFFSET) + (firing * VLP16_FIRING_TOFFSET)) /
-                 VLP16_BLOCK_DURATION);
+                 VLP16_BLOCK_DURATION) - corrections.rot_correction * 180.0 / M_PI * 100;
+          
+              if (azimuth_corrected_f < 0.0){
+                // std::cout << "azimuth_corrected_f: " << azimuth_corrected_f << std::endl;
+                azimuth_corrected_f += 36000.0;
```
Comment (mojomex) [inline nebula_decoders/src/nebula_decoders_velodyne/decoders/vlp16_decoder.cpp:238]:
```suggestion
              if (azimuth_corrected_f < 0.0){
                azimuth_corrected_f += 36000.0;
```

**Comment by vividf** [comment]
I checked the azimuth value and LGTM!

**Review by vividf — APPROVED** [review]
I checked the azimuth value with atan2(y, x) and LGTM!

File: src/ros2_socketcan
Comment (mojomex) [inline src/ros2_socketcan:1]: Please remove this from your PR

File: src/transport_drivers
Comment (mojomex) [inline src/transport_drivers:1]: Please remove this from your PR

**Review by mojomex — CHANGES_REQUESTED** [review]
Everything looks good, but could you quickly remove the two submodules that were erroneously added?
Thank you!

**Comment by mojomex** [comment]
Hi, I just merged the current `main` branch into your PR but it seems that this PR:
- https://github.com/tier4/nebula/pull/150
produces a conflict in the VLS128 unit tests (that PR modified the Rosbag used, so the unit tests of this PR fail now).

Could you regenerate the VLS128 unit test `.pcd` file once more?
Sorry for the extra work this causes :bowing_man:

**Review by mojomex — APPROVED** [review]
I took the liberty to remove `transport_drivers` again :wink:
LGTM!

**Pull request merged**